File: threedollar-api/src/main/java/com/depromeet/threedollar/api/service/store/dto/response/StoreInfoResponse.java
package com.depromeet.threedollar.api.service.store.dto.response;

import com.depromeet.threedollar.application.common.dto.AuditingTimeResponse;
import com.depromeet.threedollar.domain.domain.menu.MenuCategoryType;
import com.depromeet.threedollar.domain.domain.store.Store;
import lombok.*;

import java.util.ArrayList;
import java.util.List;

@ToString
@Getter
@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class StoreInfoResponse extends AuditingTimeResponse {

    private Long storeId;
    private double latitude;
    private double longitude;
    private String storeName;
    private double rating;
    private Boolean isDeleted;
    private final List<MenuCategoryType> categories = new ArrayList<>();

    @Builder(access = AccessLevel.PRIVATE)
    private StoreInfoResponse(Long storeId, double latitude, double longitude, String storeName, double rating, Boolean isDeleted) {
        this.storeId = storeId;
        this.latitude = latitude;
        this.longitude = longitude;
        this.storeName = storeName;
        this.rating = rating;
        this.isDeleted = isDeleted;
    }

    public static StoreInfoResponse of(Store store) {
        StoreInfoResponse response = StoreInfoResponse.builder()
            .storeId(store.getId())
            .latitude(store.getLatitude())
            .longitude(store.getLongitude())
            .storeName(store.getName())
            .rating(store.getRating())
            .isDeleted(store.isDeleted())
            .build();
        response.categories.addAll(store.getMenuCategoriesSortedByCounts());
        response.setBaseTime(store);
        return response;
    }

}

File: threedollar-admin/src/main/kotlin/com/depromeet/threedollar/admin/service/store/dto/response/StoreInfoResponse.kt
package com.depromeet.threedollar.admin.service.store.dto.response

import com.depromeet.threedollar.application.common.dto.AuditingTimeResponse
import com.depromeet.threedollar.domain.user.domain.store.MenuCategoryType
import com.depromeet.threedollar.domain.user.domain.store.Store

data class StoreInfoResponse(
    val storeId: Long,
    val latitude: Double,
    val longitude: Double,
    val storeName: String,
    val rating: Double,
    val categories: List<MenuCategoryType>
) : AuditingTimeResponse() {

    companion object {
        fun of(store: Store): StoreInfoResponse {
            val response = StoreInfoResponse(
                store.id,
                store.latitude,
                store.longitude,
                store.name,
                store.rating,
                store.menuCategoriesSortedByCounts
            )
            response.setBaseTime(store)
            return response
        }
    }

}
